### PR TITLE
Mario blanco/modal brave

### PR DESCRIPTION
**Pull request by MarioBlanco**
### Description:

As its explained in the issue #1816 , the DApp does not work while using Brave unless they disable their shields.

Please explain the changes you made here:

· Crated a modal to warn the user that he has to lower his shields to avoid problems with OGN dApp.
· Modify App.js to include the detection and show the modal in the case that Brave is detected.

- A description of the problem you're trying to solve
Brave is a browser that tries to go undetected. That's why a simple analysis of the user agent or plugins used by the client isn't enough to actually detect Brave without false positives.

- An overview of the suggested solution

it happens that it's possible to detect brave though the duckduckgo API, i made an async request to the duckduckgo api and check if the brave word is present in the requested useragent.
 
- If the feature changes current behavior, reasons why your solution is better
currently, no detection exists so no behavior change here.

### Checklist:

- [x] Test your work and double-check to confirm that you didn't break anything
- [x] [Wrap any new text/strings with `fbt`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#wrapping-text) for translation
- [x] If there are any new/changed strings to translate, [`npm run translate`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#integrating-translations)
- [x] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/docs)

### CODE REVIEW

**Comment by MarioBlanco** [comment]
Now if the user is using brave, he will see the mobile modal after he dismisses the warning modal. If the user is not using brave he will see the mobile modal as expected by the standard behavior

File: dapps/marketplace/src/pages/App.js [hunk 1]
```diff
@@ -60,12 +71,27 @@ class App extends Component {
...
+    }
   }
 
   static getDerivedStateFromError(err) {
     return { hasError: true, err }
   }
 
+  async checkBrave() {
+    const isBrave = (await (await fetch(
+      'https://api.duckduckgo.com/?q=useragent&format=json'
```
[the remark this answers is not in the document]
Comment (franckc) [reply dapps/marketplace/src/pages/App.js]: As opposed to relying on a yet another centralized r3rd party service, could we possibly do the Brave browser detection ourselves ?

I wonder for example if we could use the technique mentioned in this article https://www.ctrl.blog/entry/brave-user-agent-detection.html
It might yield some false positive on mobile so we would have to test a bit...

[the remark this answers is not in the document]
Comment (MarioBlanco) [reply dapps/marketplace/src/pages/App.js]: I just checked the proposed code on that article and it doesn't detect brave (at least in my tests).
That's probably because of many versions of Brave ahead of that solution. 
Before using duckduckgo proposed solution, I did tests using the user agent variable, the number of loaded extensions, etc to try and detect brave consistently. It happens to be very hard to detect.
I'm sure we can implement a local detection for the current version of Brave if time is assigned to the task, I can do it if you will. But bear in mind that we will be merging a solution that may fail tomorrow or the next day, as soon as Brave release a new version (they usually change behaviors to keep blending with chrome).

I know that its kind of funky to rely on a 3rd party on this issue, but it seems duckduckgo keeps the Brave detection updated. So your call, reopen the issue if we prefer to implement a local detection, and i will look into it.

[the remark this answers is not in the document]
Comment (sparrowDom) [reply dapps/marketplace/src/pages/App.js]: Reading a bit about the Brave browser and its detection, it seems that this is a bit of a cat and mouse game. There are thing that have been working years/months ago and do not right now. Seems that brave is actively trying to fight pages detecting their browser. 

@franckc the solution from the article does not work for me either.

@MarioBlanco If you want to give it a go for a local solution I think this is a good place to start: https://github.com/brave/brave-browser/issues/5768 I have tested it and it was working for me on mobile/chrome/brave browser without false positives. I only haven't implemented the logic for detection of ad blocking. - have hardcoded it while testing.

If you want to finish it this is the branch where functionality except ad blocking is implemented: https://github.com/OriginProtocol/origin/pull/3177 There is also some cleanup how UI modals display.

[the remark this answers is not in the document]
Comment (MarioBlanco) [reply dapps/marketplace/src/pages/App.js]: Ok, nice cleanup. Going to implement adblocking detection as you suggested.

[the remark this answers is not in the document]
Comment (sparrowDom) [reply dapps/marketplace/src/pages/App.js]: @MarioBlanco for now the brave modal stuff has been removed. From my testing experience the DApp works in Brave with default settings. And if you disable cookies it totally breaks. So the modal would never display. You can read more about it here:

https://github.com/OriginProtocol/origin/pull/3177

If you have a different experience, or think of a better solution than suggested in the linked PR please share it :)

**Comment by tomlinton** [comment]
Just muddying the waters a bit! Perhaps we are approaching this from the wrong angle. The issue here is that Brave disables certain functionality (probably cookies, not sure about local storage, etc). What if instead of detecting Brave, we instead focus on detecting the required functionality? It's just a more generic approach to the same thing.

We then can display a message like "It looks like your browser does not support cookies. If you are using Brave you may need to take your shields down."

**Comment by sparrowDom** [comment]
Yup good point @tomlinton . There is a comment proposing the same thing in the PR that was reverting this one: https://github.com/OriginProtocol/origin/pull/3177

**Comment by sparrowDom** [comment]
@tomlinton I see that @nick already made an issue for this: https://github.com/OriginProtocol/origin/issues/3096

**Comment by tayvano** [comment]
Sorry to hop in randomly. I think you already arrived at this so feel free to ignore me.

In this case I would def recommend figuring out what is causing Brave to block it, test that, and then show a pop-up if so/not. This will have the added benefit that users with certain extensions or settings in other browsers will still receive a modal if the feature isn't available. It will also have the benefit that if Brave edits default settings / mechanism you are using to detect Brave, you don't have to change anything.

Basically, it's not *necessarily* about Brave—it's about not access to X feature.

**Comment by micahalcorn** [comment]
Thanks @tayvano 👊 Nice to see you here!

**Comment by sparrowDom** [comment]
@tayvano great comment thanks 🙏 . I think @MarioBlanco is implementing similar solution to your proposal (https://github.com/OriginProtocol/origin/issues/3096). 

You are very welcome to weigh in to the linked issue in case you want to.
